formats/src/main/scala/org/scanamo/DerivedDynamoFormat.scala

```diff
@@ -35,75 +34,74 @@ trait DerivedDynamoFormat {
     fieldWitness: Witness.Aux[K]
   ): ValidConstructedDynamoFormat[FieldType[K, V] :: T] =
     new ValidConstructedDynamoFormat[FieldType[K, V] :: T] {
-      def read(av: AttributeValue): Validated[InvalidPropertiesError, FieldType[K, V] :: T] = {
-        val fieldName = fieldWitness.value.name
-
-        val possibleValue =
-          av.getM.asScala.get(fieldName).map(headFormat.value.read).orElse(headFormat.value.default.map(Either.right))
+      private final val fieldName = fieldWitness.value.name
 
-        val valueOrError = possibleValue.getOrElse(Either.left[DynamoReadError, V](MissingProperty))
+      final def read(av: DynamoObject) = {
+        val valueOrError: Either[(FieldName, DynamoReadError), V] =
+          av(fieldName).map(headFormat.value.read) orElse
+            headFormat.value.default.map(Either.right) getOrElse
+            Either.left(MissingProperty) leftMap
+            ((fieldName, _))
```

Comment:
I have never been a huge fan of mixing dot and space notation. Especially in the same function or even the same line